frontend/lobby.js

```diff
@@ -0,0 +1,66 @@
+// this script is loaded within an iframe, so grab the websocket from the parent
+var websocket = parent.websocket;
+
+websocket.onmessage = function(message){
+  console.log("lobby onmessage: " + message.data);
+  handleEvent(JSON.parse(message.data));
+};
+
+// websocket has alreayd been opened at this point
+var gameRequest = {eventType: "GAMES_REQUEST"};
+websocket.send(JSON.stringify(gameRequest));
+
+function createGame() {
+  var name = document.getElementById("createName").value;
+  document.getElementById("createName").value = "";
+
+  websocket.send(JSON.stringify({eventType: "CREATE_GAME_REQUEST", name: name}));
+}
+
+function joinGame(name) {
+  parent.tag = prompt("Please enter your name");
+  parent.goto("game.html");
+  var joinRequest = {eventType: "JOIN_REQUEST", playerTag: parent.tag, game: name };
+  websocket.send(JSON.stringify(joinRequest));
```

Comment:
potential issue here: if the join_notification responses from the server all come before game.html/js has fully loaded, the notifications might be lost.

store the game name in parent, and then sending a join request in game.js